### DISCUSSION TITLE
AI Conversation within Fabric?

### DISCUSSION THREAD

**Discussion by joesvetz** [danielmiessler/Fabric#437] [Q&A]
Hi Everyone! Is there any way to have a back-and-forth conversation with AI within fabric a-la ChatGPT? As of now, I've only been able to do a single question-response pair with no follow-up questions.

**Comment by digitalw00t** [comment]
echo "Hello there"|fabric --session BOBO -sp ai

This would start the conversation.

echo "I'm fine and you?"|fabric -S BOBO -sp ai

It would resume where your last prompt left off with the ai's response in the context.

WARNING:
This just keeps adding all the previous chats in the context window and can get HUGE faste.  It is not a chat bot, so it doesn't do context management/summarization nothing.  And every message will be added recursively.

Example:
USER: Hi
AGENT: Hi there how are you
CONTEXT: USER: Hi AGENT: Hi there how are you
----
USER: I'm fine and you?
AGENT: That's good to heat
CONTEXT: USER: Hi AGENT: Hi there how are you USER: I'm fine and you?
 AGENT: That's good to heat
 
 And the next context would have:
 USER: Hi AGENT: Hi there how are you USER: I'm fine and you?
 AGENT: That's good to heat REPONSES

**Reply by joesvetz** [reply]
Thanks! I look forward to giving it a try.
